### PR TITLE
[RG-311] Move save button to pin planner panels

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
 - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
 - [x] Breaking new feature. If so, please describe details in the description part.

### Describe the technical details
Move save button to pin planner panels:
image
image

Both panels have the same button. And this button is synchronized between panels.

 ### Which part of the code base require a change
 
 - [x] Library: foedagcore
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [ ] Regression tests
 - [ ] Continous Integration (CI) scripts

### CODE REVIEW

File: src/MainWindow/main_window.cpp [hunk 1]
```diff
@@ -345,6 +345,19 @@ void MainWindow::addPinPlannerRefreshButton(QDockWidget* dock) {
...
   auto layout = new QHBoxLayout;
   layout->addWidget(new QLabel{dock->windowTitle()});
+
+  auto saveButton = new QPushButton{dock};
+  saveButton->setObjectName("saveButton");
+  connect(saveButton, &QPushButton::clicked, this,
+          &MainWindow::saveActionTriggered);
+  saveButton->setSizePolicy(
+      QSizePolicy{QSizePolicy::Maximum, QSizePolicy::Maximum});
+  saveButton->setIcon(QIcon{":/images/save-action.png"});
```
Comment (ravic-rs) [inline src/MainWindow/main_window.cpp:355]: Can't we use the same save icon? Also, where did we get this from?

**Review by ravic-rs — COMMENTED** [review]
1 comment to address.

File: src/MainWindow/main_window.cpp (hunk 1 above)
Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp:355]: I've got it form https://iconoir.com/. I don't know where we got the previous. I think it make sense to replace it.

Comment (ravic-rs) [reply src/MainWindow/main_window.cpp:355]: We will have to bring in it's github codebase as a third-party module, let's talk offline.

Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp:355]: Do you want me to use same icon?

Comment (ravic-rs) [reply src/MainWindow/main_window.cpp:355]: No

Comment (alain-rs) [reply src/MainWindow/main_window.cpp:355]: A reference to the source is also acceptable.

Comment (ravic-rs) [reply src/MainWindow/main_window.cpp:355]: @KochynVolodymyr lets add a link here for reference as Alain proposed and will merge in later tonight. I will work offline to get other licenses pulled in.

Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp:355]: Link: data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSIyNHB4IiBoZWlnaHQ9IjI0cHgiIHN0cm9rZS13aWR0aD0iMS41IiB2aWV3Qm94PSIwIDAgMjQgMjQiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgY29sb3I9IiMyMDRhODciPjxwYXRoIGQ9Ik0zIDcuNVY1YTIgMiAwIDAxMi0yaDExLjE3MmEyIDIgMCAwMTEuNDE0LjU4NmwyLjgyOCAyLjgyOEEyIDIgMCAwMTIxIDcuODI4VjE5YTIgMiAwIDAxLTIgMkg1YTIgMiAwIDAxLTItMnYtMi41TTYgMjF2LTQiIHN0cm9rZT0iIzIwNGE4NyIgc3Ryb2tlLXdpZHRoPSIxLjUiIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIgc3Ryb2tlLWxpbmVqb2luPSJyb3VuZCI+PC9wYXRoPjxwYXRoIGQ9Ik0xOCAyMXYtNy40YS42LjYgMCAwMC0uNi0uNkgxNU0xNiAzdjUuNGEuNi42IDAgMDEtLjYuNmgtMS45TTggM3YzTTEgMTJoMTFtMCAwTDkgOW0zIDNsLTMgMyIgc3Ryb2tlPSIjMjA0YTg3IiBzdHJva2Utd2lkdGg9IjEuNSIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIj48L3BhdGg+PC9zdmc+

Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp:355]: Here is link for download:
data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSIyNHB4IiBoZWlnaHQ9IjI0cHgiIHN0cm9rZS13aWR0aD0iMS41IiB2aWV3Qm94PSIwIDAgMjQgMjQiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgY29sb3I9IiMyMDRhODciPjxwYXRoIGQ9Ik0zIDE5VjVhMiAyIDAgMDEyLTJoMTEuMTcyYTIgMiAwIDAxMS40MTQuNTg2bDIuODI4IDIuODI4QTIgMiAwIDAxMjEgNy44MjhWMTlhMiAyIDAgMDEtMiAySDVhMiAyIDAgMDEtMi0yeiIgc3Ryb2tlPSIjMjA0YTg3IiBzdHJva2Utd2lkdGg9IjEuNSI+PC9wYXRoPjxwYXRoIGQ9Ik04LjYgOWg2LjhhLjYuNiAwIDAwLjYtLjZWMy42YS42LjYgMCAwMC0uNi0uNkg4LjZhLjYuNiAwIDAwLS42LjZ2NC44YS42LjYgMCAwMC42LjZ6TTYgMTMuNlYyMWgxMnYtNy40YS42LjYgMCAwMC0uNi0uNkg2LjZhLjYuNiAwIDAwLS42LjZ6IiBzdHJva2U9IiMyMDRhODciIHN0cm9rZS13aWR0aD0iMS41Ij48L3BhdGg+PC9zdmc+

Comment (ravic-rs) [reply src/MainWindow/main_window.cpp:355]: Thanks. I will work on integrating that github codebase separately, will merge this in.

**Pull request merged**